src/v/rpc/simple_protocol.cc

```diff
@@ -33,6 +33,9 @@ struct server_context_impl final : streaming_context {
     ~server_context_impl() override { res.probe().request_completed(); }
     const header& get_header() const final { return hdr; }
     void signal_body_parse() final { pr.set_value(); }
+    void body_parse_exception(const std::exception_ptr& e) final {
+        pr.set_exception(e);
```

Comment:
Exception ptr can be copied. Don’t you want to move the exception?

Reply:
done

Reply:
what is the point of std::move(e) here? exception_ptr is trivially copyable

Reply:
`exception_ptr` has semantics similar to `shared_ptr`, there is no need to copy it.